### PR TITLE
Added buildable instructions for Rivanna

### PR DESCRIPTION
**Pull request by arupcsedu**
Signed-off-by: Arup Sarker <arupcsedu@gmail.com>

### CODE REVIEW

File: python/pycylon/test/test_data_split.py [hunk 1]
```diff
@@ -61,4 +61,8 @@ def test_data_split():
     env.barrier()
     
     if env.rank == 0:
-        os.remove(data_file)
+        try:
+            os.remove(data_file)
+            return False
+        except OSError:    # file is in use
+            return True
```
Comment (vibhatha) [inline python/pycylon/test/test_data_split.py:68]: I assume you're trying to remove the data after doing the test? If that is the case, it is better to leverage `pytest` and use a temporary directory which gets cleaned up by the test framework itself. Looking into the exising code, previous please can also be cleaned up. 

https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

File: rivanna/README.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
+# Running Cylon on Rivanna
+
+Arup Sarker (arupcsedu@gmail.com, djy8hg@virginia.edu)
```
Comment (vibhatha) [inline rivanna/README.md:3]: Generally don't put author names unless it is a blog post or an article. But I am not very much informed with the recent practices, so I could be wrong. :slightly_smiling_face:

File: rivanna/README.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  - fsspec>=0.6.0
+  - setuptools
+  # they are not needed for using pygcylon or compiling it
+  - pytest
+  - pytest-mpi
+  - mpi4py
+  - gcc
+  - gxx
+  - gxx_linux-64
+
```
Comment (vibhatha) [inline rivanna/README.md:59]: @arupcsedu just a suggestion, would it be nice to keep this in a `requirements.txt` and just mention the usage in the `README.md`. This way, we can keep minimal edits to the markdown file but add dependencies to the `requirements.txt`.

File: rivanna/README.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  - gcc
+  - gxx
+  - gxx_linux-64
+  
+ ```
+
+Added all blue packages to overcome the dependencies on the preloaded modules.
+
+```shell
+conda env create -f conda/environments/cylon.yml
```
Comment (vibhatha) [inline rivanna/README.md:65]: Should we specify a `Python` version for this?

File: rivanna/README.md (hunk 3 above)
Comment (vibhatha) [reply rivanna/README.md:59]: And if users has additional dependencies, they can create `requirements_extras.txt` to include their own dependencies but use our `requirements.txt` as the base dependencies.

**Comment by vibhatha** [comment]
@arupcsedu Thanks for working on this. I have externally looked at this.  Sorry about my lack of knowledge about the context of the PR. It would be better to include a good description in the PR to give some clarity on what the feature is introducing. And later on when we run into issues, we could rely on this description and debug. 

I assume this is a platform or compute resource where we want to setup Cylon. Please correct me if I am wrong.

Overall the PR looks good to me in external shape, but I haven't and don't know how to test the instructions to see if they function properly. 

Is there a way to include a CI to Cylon to run this with the said platform/resources? If not, it would be better to keep an isolated/internal workload which can test this. 

Would it be wise to include a docker or a singularity image which could be deployed quite easily?

**Comment by laszewsk** [comment]
* docker is not available in rivanna
* my experience from using docker to singularity to rivanna is miserable due to very slow upload tmes.
* singularity has very limited images available on rivanna from my experience it is best tho have a modified requirements such as the one discussed here.
* As the setup Is so complex it is advisable to distribute this setup so that it can also serve as example for others.
* Maybe you want give all platforms a special directory as i imagine this could be an issue for others.
* one issue for adaption of cylon is platform differences. please remember we have a similar issue for summit.

The conclusion I have is that each platform needs a special documentation so that other do not spend hours, days or weeks for the deployment and they can learn from each other

File: rivanna/README.md (hunk 3 above)
Comment (laszewsk) [reply rivanna/README.md:59]: I think its better to not have a requirements_extra.txt but a 
requirements-rivanna.txt
The reason is that on other platforms you may have different versions of non python libraries as default, and you want to make sure that default software provided by the rivanna system staff can be used by having an explicit rivanna requirements.txt.
Also as this is conda, should it not be in something like
conda/environmenst/rivanna/cylon.yaml
?

File: python/pycylon/test/test_data_split.py (hunk 1 above)
Comment (arupcsedu) [reply python/pycylon/test/test_data_split.py:68]: os.remove(data_file) was used before and caused an intermittent error during running on a windows machine. 

I just added the code with exception handling. This issue is happening only in this test case. So, we should not make it standard for all other test cases.

File: rivanna/README.md (hunk 2 above)
Comment (arupcsedu) [reply rivanna/README.md:3]: I followed this convention by taking the sample from Summit compatibility documents. 
https://github.com/cylondata/cylon/tree/main/summit#readme

File: rivanna/README.md (hunk 3 above)
Comment (arupcsedu) [reply rivanna/README.md:59]: @vibhatha, This is the readme file and I tried to make minimal changes to the existing default build yaml file.
This are the additional default packages:   
- gcc
- gxx
- gxx_linux-64
are needed on top of existing packages.

It is not ensured that these packages will be the same after 2 years.  During that time, there can build errors. 
Moreover, I have found another way to make Cylon compatible with Rivanna. So, developers have to maintain it and might get confused.  We tried to make it as simple as possible. That's why after discussing with @nirandaperera , we have decided to make an additional readme file.

File: rivanna/README.md (hunk 4 above)
Comment (arupcsedu) [reply rivanna/README.md:65]: We don't need it. Rivanna's default python version works perfectly.

File: rivanna/README.md (hunk 3 above)
Comment (arupcsedu) [reply rivanna/README.md:59]: @laszewsk , we thought to create a different yaml file for Rivanna. But we have two solutions and both solution needs minimal different changes on the default cylon yaml file. So, we kept both options open for developers.

**Review by nirandaperera — APPROVED** [review]

**Pull request merged**